File: jan-bronze-2021/uddered.java
import java.util.Scanner;

public class uddered {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String alphabet = sc.nextLine();
        String letters = sc.nextLine();

        int last = -1;
        int times = 1;

        for (int i=0; i<letters.length(); i++) {
            int index = alphabet.indexOf(letters.charAt(i));
        
            if (index <= last) {
                times++;
            }

            last = index;
        }

        System.out.println(times);

        sc.close();
    }
}

File: jan-bronze-2021/uddered.cpp
#include <iostream>
using namespace std;
#include <string>
 
int main() {
    // example 
    // abcdefghijklmnopqrstuvwxyz
    // mood
    string alphabet; getline(cin, alphabet);
    string letters; getline(cin, letters);
    int last = -1;
    int times = 1;

    // loop through letters
    for (int i=0; i<letters.length(); i++) {
        int index = alphabet.find(letters[i]);
        
        if (index <= last) {
            times++;
        }

        last = index;
    }
    
    cout << times;
}
